cardano-crypto-class/src/Cardano/Crypto/KES/Simple.hs

```diff
@@ -12,7 +12,9 @@
 {-# LANGUAGE NoStarIsType #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 
--- Needed for ghc-9.6.
+-- Needed for ghc-9.6 to avoid a redunant constraint warning on the
+-- `KESSignAlgorithm m (SimpleKES d t)` instance. Removing the constraint leaves another type
+-- error which is rather opaque.
 {-# OPTIONS_GHC -Wno-redundant-constraints #-}
```

Comment:
@lehins maybe something you can take a look at when you are having a free minute?

Reply:
Sounds good, I'll take a look at it.

I've seen similar behavior in one of my libraries with ghc-9.4, which complains about redundant constraint, but removing the constraint breaks ghc-8

So, we might have to add CPP for the GHC flag in order to accommodate older ghc versions